packages/launchpad-ledger/src/launchpadledger.ts

```diff
@@ -0,0 +1,232 @@
+    hdPath: number[] = cosmosHdPath,
+    prefix: string = cosmosBech32Prefix,
+  ) {
+    this.testModeAllowed = testModeAllowed;
+    this.hdPath = hdPath;
+    this.prefix = prefix;
+    this.cosmosApp = null;
+    this.platform = navigator.platform;
+    this.userAgent = navigator.userAgent;
+  }
+
+  // // quickly test connection and compatibility with the LaunchpadLedger device throwing away the connection
+  // async testDevice(): Promise<LaunchpadLedger> {
+  //   // poll device with low timeout to check if the device is connected
+  //   const secondsTimeout = 3; // a lower value always timeouts
+  //   await this.connect(secondsTimeout);
+  //   this.cosmosApp = null;
+
+  //   return this;
+  // }
+
+  async connect(timeout = defaultInteractionTimeout): Promise<LaunchpadLedger> {
+    // assume good connection if connected once
+    if (this.cosmosApp) {
+      return this;
+    }
+
+    verifyBrowserIsSupported(this.platform, this.userAgent);
+
+    const transport = await createTransport(timeout * 1000);
+    this.cosmosApp = new CosmosApp(transport);
+
+    await this.verifyDeviceIsReady();
+    return this;
+  }
+
+  async getCosmosAppVersion(): Promise<string> {
+    await this.connect();
+    assert(this.cosmosApp, "Cosmos Ledger App is not connected");
+
+    const response = await this.cosmosApp.getVersion();
+    this.handleLedgerErrors(response);
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { major, minor, patch, test_mode: testMode } = response as VersionResponse;
+    this.verifyAppMode(testMode);
+    return `${major}.${minor}.${patch}`;
+  }
+
+  async getPubKey(): Promise<Buffer> {
+    await this.connect();
+    assert(this.cosmosApp, "Cosmos Ledger App is not connected");
+
+    const response = await this.cosmosApp.publicKey(this.hdPath);
+    this.handleLedgerErrors(response);
+    return (response as PublicKeyResponse).compressed_pk;
+  }
+
+  async getCosmosAddress(): Promise<string> {
+    const pubKey = await this.getPubKey();
+    return CosmosApp.getBech32FromPK(this.prefix, pubKey);
+  }
+
+  // async verifyLedgerAddress(): Promise<void> {
+  //   await this.connect();
+  //   assert(this.cosmosApp, "Cosmos Ledger App is not connected");
+
+  //   const response = await this.cosmosApp.showAddressAndPubKey(this.hdPath, this.bech32Prefix);
+  //   this.handleLedgerErrors(response, {
+  //     rejectionMessage: "Displayed address was rejected by the user",
+  //   });
+  // }
+
+  async sign(message: Uint8Array): Promise<Uint8Array> {
+    await this.connect();
+    assert(this.cosmosApp, "Cosmos Ledger App is not connected");
+
+    const response = await this.cosmosApp.sign(this.hdPath, fromUtf8(message));
+    this.handleLedgerErrors(response, {
+      rejectionMessage: "Transaction signing request was rejected by the user",
+    });
+    const { r: rRaw, s: sRaw } = bip66.decode((response as SignResponse).signature);
```

Comment:
Could you try `Secp256k1Signature.fromDer((response as SignResponse).signature).toFixedLength()` with

```ts
export class Secp256k1Signature {
  // ...

  public toFixedLength(): Uint8Array {
    return new Uint8Array([...this.r(32), ...this.s(32)]);
  }
}
```

? Then we can remove the bip66 dependency.

Reply:
Brilliant, I didn't find that method.